pexconn.go

```diff
@@ -0,0 +1,116 @@
+package torrent
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/anacrolix/log"
+
+	pp "github.com/anacrolix/torrent/peer_protocol"
+)
+
+const (
+	pexRetryDelay = 10 * time.Second
+	pexInterval   = 1 * time.Minute
+)
+
+// per-connection PEX state
+type pexConnState struct {
+	enabled bool
+	xid     pp.ExtensionNumber
+	seq     int
+	timer   *time.Timer
+	gate    chan struct{}
+	readyfn func()
+	torrent *Torrent
```

Comment:
Possibly put this behind an interface?

Reply:
Without another meaningful thing to put behind the interface, it feels like it would just add clutter without any real return. I hope it makes sense.

Reply:
It does!